Draft long-distance transit assignment

zptro opened the pull request:
- Remove `l` mode (airplane) from `transit_work` and `transit_leisure` assignment
- Enable `transit_work` and `transit_leisure` assignment in `WholeDayPeriod`
  - No specific mode is enforced
- Remove separate `train` and `coach` assignment
- Introduce new joint `train` and `coach` car-access assignment
  - Either `j` or `e` mode enforced (no enforcement? `j` enforced?)

zptro committed:
Set in-vehicle time weight to 0.7 for long-distance train

zptro committed:
Aggregate far-away destinations

zptro committed:
Revert "Aggregate far-away destinations"

This reverts commit d7dfa664f91aa34b56054a5e980701fa23b2776c.

zptro committed:
Draft long-distance transit assignment

zptro committed:
Merge branch 'dev' into feat/long-dist-transit

attesn committed:
Add first demand parameters for long distance transit

attesn committed:
New demand parameters with correct signs
Specification with log time and alternative specific logsum

attesn committed:
Fix test data and mode share integration test

attesn committed:
New estimation with park cost and without weights

attesn committed:
Add car egress to access mode alternatives

attesn committed:
Add car use to access mode choice parameters

zptro committed:
Merge branch 'dev' into feat/train-inv-weight

attesn committed:
Add walk mode to overnight tours

attesn committed:
Add walk mode to integration tests

zptro committed:
Add walk and bike to long-dist assignment

zptro committed:
Fix import of airplane matrix in short-distance model

zptro committed:
Remove extra car-cost handling

Since December 2024, cost matrices always include both distance-based
and toll-based costs. Hence we no longer need to handle them
separately in mock assignment.

attesn committed:
Use network walk distance in demand parameters

attesn committed:
Merge branch 'dev' into feat/long-dist-transit

attesn committed:
Add new distance interval to size factor

attesn committed:
Estimate model parameters with generalized cost

attesn committed:
Add new distance intervals to size factor

attesn committed:
Replace size factor distance intervals with attraction vars

attesn committed:
Convert to dest-mode model structure

attesn commented:
Addition of walk to long distance assignment causes error:

```
[INFO 15:59] Pedestrian assignment started...
[EXCEPTION] Traceback (most recent call last):
[EXCEPTION] File "--\Scripts\valma_travel.py", line 341, in <module>
[EXCEPTION] main(args)
[EXCEPTION] File "--\Scripts\valma_travel.py", line 123, in main
[EXCEPTION] impedance = model.assign_base_demand(
[EXCEPTION] File "--\Scripts\travel_iteration.py", line 295, in assign_base_demand
[EXCEPTION] ap.init_assign()
[EXCEPTION] File "--\Scripts\assignment\long_dist_period.py", line 57, in init_assign
[EXCEPTION] self.walk_mode.assign()
[EXCEPTION] File "--\Scripts\assignment\datatypes\assignment_mode.py", line 189, in assign
[EXCEPTION] self.emme_project.transit_assignment(
[EXCEPTION] File "inro\director\logging\trail.py", line 279, in __trace
[EXCEPTION] File "standard\inro\emme\transit_assignment\extended_transit_assignment.py", line 261, in call
[EXCEPTION] inro.emme.core.exception.ArgumentError: class_name: already used in previously assigned classes.
```

This can be fixed by setting `add_volumes=False`, but not sure if that leads to some other problems.

zptro commented:
> Addition of walk to long distance assignment causes error:
> 
> ```
> [INFO 15:59] Pedestrian assignment started...
> [EXCEPTION] Traceback (most recent call last):
> [EXCEPTION] File "--\Scripts\valma_travel.py", line 341, in <module>
> [EXCEPTION] main(args)
> [EXCEPTION] File "--\Scripts\valma_travel.py", line 123, in main
> [EXCEPTION] impedance = model.assign_base_demand(
> [EXCEPTION] File "--\Scripts\travel_iteration.py", line 295, in assign_base_demand
> [EXCEPTION] ap.init_assign()
> [EXCEPTION] File "--\Scripts\assignment\long_dist_period.py", line 57, in init_assign
> [EXCEPTION] self.walk_mode.assign()
> [EXCEPTION] File "--\Scripts\assignment\datatypes\assignment_mode.py", line 189, in assign
> [EXCEPTION] self.emme_project.transit_assignment(
> [EXCEPTION] File "inro\director\logging\trail.py", line 279, in __trace
> [EXCEPTION] File "standard\inro\emme\transit_assignment\extended_transit_assignment.py", line 261, in call
> [EXCEPTION] inro.emme.core.exception.ArgumentError: class_name: already used in previously assigned classes.
> ```
> 
> This can be fixed by setting `add_volumes=False`, but not sure if that leads to some other problems.

Long-dist assignment differs from the other assignment types in that strategies are stored by default, so that they can be reused in network analysis after demand calculation. But if walk assignment is performed first, I think it should be safe to set `add_volumes=False` for that.

zptro commented on Scripts/assignment/emme_assignment.py:
```diff
@@ -136,7 +136,7 @@ def prepare_network(self, car_dist_unit_cost: Dict[str, float],
                 use_stored_speeds=(car_time_files is not None),
                 delete_extra_matrices=self.delete_extra_matrices,
                 delete_strat_files=self._delete_strat_files))
-        ass_classes = (param.car_classes + param.long_distance_transit_classes
+        ass_classes = (param.car_classes + param.transit_classes
             if self.use_free_flow_speeds else param.simple_transport_classes)
```
@attesn Here is the extra-attribute creation

jalovaj2 committed:
Skip deletion of non-existing distance matrices (#333)

jalovaj2 committed:
Merge remote-tracking branch 'origin/feat/train-inv-weight' into feat/long-dist-transit

attesn commented:
It seems that `walk` matrices are not exported to `dist_vrk.omx` matrices from assignment. @zptro

zptro commented on Scripts/assignment/long_dist_period.py:
```diff
@@ -107,23 +115,17 @@ def end_assign(self,
             self._calc_transit_link_results()
-        return self._get_impedances(param.car_classes
-                                    + param.long_distance_transit_classes)
+        return self._get_impedances(param.car_classes + param.transit_classes)
```
`walk` has been a bit of an exception in assignment, the easiest way of fixing its output would be to add `+ ("walk",)` here, but I admit it is not a very nice solution @attesn

attesn replied:
Could you fix this and bike attribute above? We'll have strict deadline this week and I've not done any changes to assignment before. There's quite a lot of code here to comprehend.

zptro replied:
Fixed

zptro committed:
Update impedance calculation to include walking class

zptro committed:
Add 'bike' to extra attributes

zptro committed:
Fix syntax error

jalovaj2 committed:
Merge branch 'feat/long-dist-transit' of https://github.com/Traficom/lem-model-system into feat/long-dist-transit

attesn committed:
Calibrate assignment parameters for long distance transit (#340)

* Calibrate boarding and transfer penalties

* Calibrate mode choice constants
Match mode shares to survey after assignment parameter changes

* Update assignment parameters

* Remove coach from car access and egress modes
Park cost is not defined for coach so it cannot be assigned

attesn committed:
Merge branch 'dev' into feat/long-dist-transit

attesn committed:
Remove train and coach parameters

attesn reviewed (approved):
Demand parameters are better in this version. We were not able to validate assignment due to very time consuming model runs and multiple errors.

I think we'll merge this PR and continue assignment parameter development after we've optimized model runs.
